### PR TITLE
[FEAT] Refactor the layout of modules

### LINKED ISSUE #3485: Enhance OverviewActions to have custom buttons
`OverviewActions` component is used in all **overview dashboards** and some of them **don't need an agent selector**. This should be optional to meet each module needs.

image

**Additional context**

### PR DESCRIPTION
**Pull request by Desvelao**
### Changes
- Refactor `modules-defaults.js` to define what buttons and components are rendered in each module tab.
- Removed old logic to render the component depending on the selected module and tab.
- Removed the `Explore agent` button from `Modules/MITRE ATT&CK/Intelligence`.
- Created `PromptModuleNotForAgent` component.
- Created `withModuleNotForAgent` HOC.

### Tests
- Review each tab for each module is rendering its view.
- Review the `Explore agent` is working as usual.
- Review the `Generate report` is working as usual.
- Review the `Modules/MITRE ATT&CK/Intelligence` is not displaying the `Explore agent` button.

Closes #3485

### CODE REVIEW

**Comment by pablomarga** [comment]
When generation a report in Security Events this error is displayed in the dev tools console
Screenshot_20210721_131731

**Review by sortizowlh — APPROVED** [review]
CR & Test: LGTM

File: public/components/agents/prompts/prompt_module_not_for_agent.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  body?: string;
+  title: string;
+  agentSelectionProps: {
+    setAgent: (agent: boolean) => void
+  }
+};
+
+export const PromptModuleNotForAgent = ({ body, title, ...agentSelectionProps }: PromptSelectAgentProps) => {
+  const dispatch = useDispatch();
+  const filterManager = useFilterManager();
```
Comment (mpRegalado) [inline public/components/agents/prompts/prompt_module_not_for_agent.tsx:29]: This has changed in the office365 branch. Noting it so we are mindful when merging there.

**Review by mpRegalado — APPROVED** [review]
CR: LGTM :heavy_check_mark:

**Comment by CPAlejandro** [comment]
**Changes applied:**
- In `visualizationIDList` we save the visualizations that we are going to show, thus avoiding the undefined that were previously passed to it in item.

File: public/components/common/hocs/with_module_not_for_agent.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
```
[the remark this answers is not in the document]
Comment (Desvelao) [reply public/components/common/hocs/with_module_not_for_agent.tsx:12]: Fixed

**Review by mpRegalado — APPROVED** [review]
CR: LGTM :heavy_check_mark:

**Comment by Desvelao** [comment]
### Changes
- Updated the branch with the last changes of `4.3-7.10`
- Resolved the conflicts

**Pull request merged**